Makefile

```diff
@@ -65,8 +67,37 @@ else
 	echo "tls disabled."
 endif
 
+CONTROLLER_GEN=$(PROJECT_DIR)/bin/controller-gen
+# find or download controller-gen
+# download controller-gen if necessary
+$(CONTROLLER_GEN):
```

Comment:
Why do we need a make target with a dynamic name?

Reply:
well, actually it is only the base path set to the current project base directory.

I guess it will also work like this:

```
bin/controller-gen:
     ....
```

But I would use the absolute path, instead of relative just in case current working directory is not the same as the authorino project directory

Reply:
What do you think about this? Makefiles usually assume your CWD is the one which has the makefile, so I guess, it should be safe to do targets like `bin/controller-gen`

Reply:
targets defined as 
```
bin/kustomize:
     ....
bin/controller-gen:
     .....
```